db/config.hh

```diff
@@ -532,6 +553,7 @@ public:
     const db::extensions& extensions() const;
 
     utils::updateable_value_source<std::unordered_map<sstring, s3::endpoint_config>> object_storage_config;
```

Comment:
This dedicated map of endpoints should be removed, and all the code that still uses it should be switched to be using the newly introduced object_storage_endpoints

Reply:
I'd do this in a separate PR so we get this one out and people can use it, you ok with this?

Reply:
No, please, do it right at once, in this PR